frontend/app/src/components/helper/MenuTooltipButton.vue

```diff
@@ -1,29 +1,49 @@
 <script setup lang="ts">
-defineProps({
-  tooltip: { required: true, type: String, default: '' },
-  onMenu: { required: false, type: Object, default: () => {} },
-  retainFocusOnClick: { required: false, type: Boolean, default: false },
-  className: { required: false, type: String, default: '' }
-});
+withDefaults(
+  defineProps<{
+    tooltip: string;
+    onMenu?: object;
+    retainFocusOnClick?: boolean;
+    className?: string;
+    href?: string;
+  }>(),
+  {
+    onMenu: undefined,
```

Comment:
thought the default for object and array will be a function that returns the value.
like `() => []`  for array and `() => ({})` for object

Reply:
oh, no worries, since you set to `undefined` and not `{}`